Scheduled notifications result in exception "Can't infer the SQL type to use for an instance of org.datanucleus.store.types.wrappers.Date. Use setObject() with an explicit Types value to specify the type to use"

### Current Behavior

When configuring scheduled/summarized notifications we have noted the following warning in the logs and that summarized notifications are not sent:

`
2025-04-08 12:14:00,780 WARN [ScheduledNotificationDispatchTask] Failed to dispatch notification for group NEW_VULNERABILITIES_SUMMARY [notificationRuleUuid=2afdae85-96ae-4b8e-9761-f940f436f498, notificationRuleName=slack-test]
javax.jdo.JDODataStoreException: Error executing SQL query "SELECT "COMPONENT"."PROJECT_ID" AS "projectId"      , "COMPONENT"."ID" AS "componentId"      , "COMPONENTS_VULNERABILITIES"."VULNERABILITY_ID" AS "vulnabilityId"      , "FINDINGATTRIBUTION"."ANALYZERIDENTITY" AS "analyzerIdentity"      , "FINDINGATTRIBUTION"."ATTRIBUTED_ON" AS "attributedOn"      , "FINDINGATTRIBUTION"."REFERENCE_URL" AS "referenceUrl"      , "ANALYSIS"."STATE" AS "analysisState"      , "ANALYSIS"."SUPPRESSED" AS "isSuppressed"   FROM "COMPONENTS_VULNERABILITIES"  INNER JOIN "COMPONENT"     ON "COMPONENT"."ID" = "COMPONENTS_VULNERABILITIES"."COMPONENT_ID"  INNER JOIN "FINDINGATTRIBUTION"     ON "FINDINGATTRIBUTION"."COMPONENT_ID" = "COMPONENT"."ID"    AND "FINDINGATTRIBUTION"."VULNERABILITY_ID" = "COMPONENTS_VULNERABILITIES"."VULNERABILITY_ID"   LEFT JOIN "ANALYSIS"     ON "ANALYSIS"."COMPONENT_ID" = "COMPONENT"."ID"    AND "ANALYSIS"."VULNERABILITY_ID" = "COMPONENTS_VULNERABILITIES"."VULNERABILITY_ID"  WHERE "FINDINGATTRIBUTION"."ATTRIBUTED_ON" >= ?    AND ("COMPONENT"."PROJECT_ID" = ? OR "COMPONENT"."PROJECT_ID" = ?)".
	at org.datanucleus.api.jdo.JDOAdapter.getJDOExceptionForNucleusException(JDOAdapter.java:608)
	at org.datanucleus.api.jdo.JDOQuery.executeInternal(JDOQuery.java:456)
	at org.datanucleus.api.jdo.JDOQuery.executeResultList(JDOQuery.java:376)
	at org.dependencytrack.tasks.ScheduledNotificationDispatchTask.getNewFindingsSince(ScheduledNotificationDispatchTask.java:424)
	at org.dependencytrack.tasks.ScheduledNotificationDispatchTask.createNewVulnerabilitiesNotification(ScheduledNotificationDispatchTask.java:169)
	at org.dependencytrack.tasks.ScheduledNotificationDispatchTask.processRule(ScheduledNotificationDispatchTask.java:118)
	at org.dependencytrack.tasks.ScheduledNotificationDispatchTask.inform(ScheduledNotificationDispatchTask.java:99)
	at alpine.event.framework.BaseEventService.lambda$publish$0(BaseEventService.java:110)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
	at java.base/java.lang.Thread.run(Unknown Source)

Caused by: org.postgresql.util.PSQLException: Can't infer the SQL type to use for an instance of org.datanucleus.store.types.wrappers.Date. Use setObject() with an explicit Types value to specify the type to use.
	at org.postgresql.jdbc.PgPreparedStatement.setObject(PgPreparedStatement.java:1064)
	at com.zaxxer.hikari.pool.HikariProxyPreparedStatement.setObject(HikariProxyPreparedStatement.java)
	at org.datanucleus.store.rdbms.query.SQLQuery.performExecute(SQLQuery.java:663)
	at org.datanucleus.store.query.Query.executeQuery(Query.java:2004)
	at org.datanucleus.store.rdbms.query.SQLQuery.executeWithMap(SQLQuery.java:840)
	at org.datanucleus.api.jdo.JDOQuery.executeInternal(JDOQuery.java:437)
`

### Steps to Reproduce

1. Create a new notification using publisher = 'Slack' and trigger type = 'Schedule'
2. Set 'Enabled' to true
3. Set 'Group' to 'NEW_VULNERABILITIES_SUMMARY'
4. Set 'Limit to projects' to one or more projects
5. Set 'Cron expression' to a valid expression...as we are testing this was set to '*/10 * * * *'
6. Set ' Skip publish if unchanged' to true (noting behavior has been noted with both true/false)
7. Review logs of container and observe logs stated above

### Expected Behavior

That scheduled/summarized notifications are sent without exceptions

### Dependency-Track Version

4.13.0

### Dependency-Track Distribution

Container Image

### Database Server

PostgreSQL

### Database Server Version

16.8

### Browser

N/A

### Checklist

- [x] I have read and understand the [contributing guidelines](https://github.com/DependencyTrack/dependency-track/blob/master/CONTRIBUTING.md#filing-issues)
- [x] I have checked the [existing issues](https://github.com/DependencyTrack/dependency-track/issues) for whether this defect was already reported

--- a/src/main/java/org/dependencytrack/tasks/ScheduledNotificationDispatchTask.java
+++ b/src/main/java/org/dependencytrack/tasks/ScheduledNotificationDispatchTask.java
@@ -389,7 +389,7 @@ public class ScheduledNotificationDispatchTask implements Subscriber {
             final Date sinceAttributedOn) {
         final var projectIdCondition = new StringJoiner(" OR ", "(", ")");
         final var queryParams = new HashMap<String, Object>(projectIds.size() + 1);
-        queryParams.put("sinceAttributedOn", sinceAttributedOn);
+        queryParams.put("sinceAttributedOn", new java.sql.Timestamp(sinceAttributedOn.getTime()));
 
         int projectIdIndex = 0;
         for (final Long projectId : projectIds) {

